### ISSUE TITLE
newsboat ignores "+" and "%" characters in mpv arguments

### ISSUE THREAD

**Issue by SlidingHorn** [newsboat/newsboat#1684]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: newsboat 2.18.0 (Ubuntu

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
# Set Browser
browser "w3m"

# If yes, then all feeds will be reloaded when Newsboat starts up. This is equivalent to the -r commandline option.
refresh-on-startup yes

# Hide empty feeds
show-read-feeds yes

# Toggle whether read feeds should be shown in the feed list.
bind-key I toggle-show-read-feeds

# Unbind
unbind-key p
unbind-key n
unbind-key g
unbind-key G

# Keybinds
bind-key h quit
bind-key j down
bind-key k up
bind-key l open
bind-key N next-unread
bind-key P prev-unread
bind-key ENTER open-in-browser
bind-key g home
bind-key G end
bind-key s sort
bind-key S rev-sort

# "Open With" macros
macro m set browser "mpv -v --ontop --volume=50 --no-border --geometry=25%+100%+100% --ytdl-format='best[height<=480]' %u"; open-in-browser ; set browser "w3m %u"
macro q set browser "qutebrowser %u"; open-in-browser ; set browser "w3m %u"
macro f set browser "firefox %u"; open-in-browser ; set browser "w3m %u"

# Exit Confirmation
confirm-exit yes

# Minutes between reload
reload-time 10
auto-reload yes

# Colors
color listnormal color12 color8
color listfocus black white
color listnormal_unread default color8
color listfocus_unread black white
color background color12 color8
color info white black
color article color12 color8
```

Steps to reproduce the issue:

1. Run the above-listed mpv macro for a youtube or peertube video

2.

3.

**Other info you think is relevant**:
Relevant portion of the output:

```
[cplayer] Setting option 'geometry' = '25100100--ytdl-format=best[height<=480]' (flags = 8)                                              
[cplayer] Option geometry: invalid geometry: '25100100--ytdl-format=best[height<=480]'                                                   
[cplayer] Valid format: [W[%][xH[%]]][{+-}X[%]{+-}Y[%]] | [X[%]:Y[%]]                                                                    
[cplayer] Error parsing option geometry (option parameter could not be parsed)
[cplayer] Setting commandline option --geometry=25100100--ytdl-format=best[height<=480] failed.                                          
[cplayer]                                                                                                                                
[cplayer] Exiting... (Fatal error)
```

**Comment by dennisschagt** [comment]
I can reproduce this with newsboat 2.18.0 but it work fine in the latest newsboat release (2.24.0).

Old versions of newsboat (<=2.19.0) handle percent characters in the `browser` configuration in a special way.
Effectively, they remove every percent-sign and the first character following it.
To pass on a literal `%`, you would need to use 2 consecutive percent characters `%%`, in case of your macro:
```
macro m set browser "mpv -v --ontop --volume=50 --no-border --geometry=25%%+100%%+100%% --ytdl-format='best[height<=480]' %u" ; open-in-browser ; set browser "w3m %u" # Only works with newsboat versions <= 2.19.0
```

This behaviour has been changed in https://github.com/newsboat/newsboat/commit/24e6773ef05bb5feed4ac3781aaa3a0e2b4504e9 (included in newsboat releases since 2.20.0).
From that point onwards, percent-signs are passed on as-is (only occurrences of`%u` are replaced with the URL).

That means, when you update to a more recent newsboat version, you will have to revert to single percent signs in your macro again:
```
macro m set browser "mpv -v --ontop --volume=50 --no-border --geometry=25%+100%+100% --ytdl-format='best[height<=480]' %u" ; open-in-browser ; set browser "w3m %u" # Only works with newsboat versions >= 2.20.0
```

**Comment by Minoru** [comment]
Since 2.20+ exhibits the correct behaviour, I consider this resolved.
